src/vivarium_gates_nutrition_optimization_child/components/wasting.py

```diff
@@ -0,0 +1,857 @@
+        pop = self.population_view.get(index)
+        exposure = pop[self.dynamic_model.state_column].apply(models.get_risk_category)
+        under_six_months = (pop["age"] < data_values.WASTING.START_AGE) & (
+            pop["alive"] == "alive"
+        )
+        if under_six_months.any():
+            exposure[under_six_months] = self.static_model.exposure(
+                pop[under_six_months].index
+            )
+        return exposure
+
+
+class StaticChildWasting(Risk):
+    def __init__(self):
+        """
+        Parameters
+        ----------
+        risk :
+            the type and name of a risk, specified as "type.name". Type is singular.
+        """
+        self.risk = EntityString("risk_factor.child_wasting")
+        name = "static_child_wasting"
+        self.configuration_defaults = self._get_configuration_defaults()
+        self.exposure_distribution = self._get_exposure_distribution()
+        self._sub_components = [self.exposure_distribution]
+
+        self._randomness_stream_name = f"initial_{name}_propensity"
+        self.propensity_column_name = f"{name}_propensity"
+        self.propensity_pipeline_name = f"{name}.propensity"
+        self.exposure_pipeline_name = f"{name}.exposure"
+
+
+class WastingTreatment(Risk):
+    def __init__(self, treatment_type: str):
+        super().__init__(treatment_type)
+
+        self.previous_wasting_column = f"previous_{data_keys.WASTING.name}"
+        self.wasting_column = data_keys.WASTING.name
+
+        self.treated_state = self._get_treated_state()
+
+    ##########################
+    # Initialization methods #
+    ##########################
+
+    def _get_treated_state(self) -> str:
+        return self.risk.name.split("_treatment")[0]
+
+    #################
+    # Setup methods #
+    #################
+
+    def setup(self, builder: Builder):
+        super().setup(builder)
+        self._register_on_time_step_prepare_listener(builder)
+
+    def _get_population_view(self, builder: Builder) -> PopulationView:
+        return builder.population.get_view(
+            [self.propensity_column_name, self.previous_wasting_column, self.wasting_column]
+        )
+
+    def _register_on_time_step_prepare_listener(self, builder: Builder) -> None:
+        builder.event.register_listener("time_step__cleanup", self.on_time_step_cleanup)
+
+    ########################
+    # Event-driven methods #
+    ########################
+
+    def on_time_step_cleanup(self, event: Event):
+        pop = self.population_view.get(event.index)
+        propensity = pop[self.propensity_column_name]
+        remitted_mask = (pop[self.previous_wasting_column] == self.treated_state) & pop[
+            self.wasting_column
+        ] != self.treated_state
+        propensity.loc[remitted_mask] = self.randomness.get_draw(remitted_mask.index)
+        self.population_view.update(propensity)
+
+
+class DynamicChildWastingModel(DiseaseModel):
```

Comment:
the reason for this subclass of DiseaseModel is to pass in the propensities of the static model